tools/src/main/java/com/nedap/archie/rules/evaluation/evaluators/FunctionUtil.java

```diff
@@ -22,17 +22,22 @@ public static double castToDouble(Value value) throws FunctionCallException {
         throw new FunctionCallException("cannot cast " + value.getValue().getClass() + " to a number");
     }
 
-    public static int checkEqualLength(List<ValueList> arguments) {
-        int length = 1;
-        boolean first = true;
-        for(ValueList list:arguments) {
-            if(first && list.size() != 1) {
-                length = list.size();
-            } else  if(list.size() != 1 && list.size() != length) {
-                return -1;
+    /**
+     * All arguments should have an equal list size OR list size of allowedLength
+     * @return If list sizes equal or allowedLength return this length, otherwise return -1
+     */
```

Comment:
As this documentation is for the users of the function, you should not refer to local variables. I would suggest to just referring to 1.
```suggestion
    /**
     * All arguments should have an equal list size OR list size of 1
     * @return If list sizes equal or 1 return this length, otherwise return -1
     */
```

Reply:
That makes sense :) Thanks!